tests/test_unit.py

```diff
@@ -34,45 +35,49 @@ def test_ensure_specificity() -> None:
     # never before published version
     empty_list: list[str] = []
-    assert _incremented_ver("1", empty_list) == "1.0.0.0"
-    assert _incremented_ver("1.2", empty_list) == "1.2.0.0"
-    assert _incremented_ver("1.2.post3", empty_list) == "1.2.3.0"
+    assert _stub_ver("1", empty_list) == f"1.0.0.{STUB_V}"
+    assert _stub_ver("1.2", empty_list) == f"1.2.0.{STUB_V}"
+    assert _stub_ver("1.2.post3", empty_list) == f"1.2.3.{STUB_V}"
 
     # published greater than version spec
-    assert _incremented_ver("1.2", ["1.3.0.4"]) == "1.3.0.5"
-    assert _incremented_ver("1.1", ["1.2.0.1"]) == "1.2.0.2"
-    assert _incremented_ver("1.1", ["1.2"]) == "1.2.0.1"
-    assert _incremented_ver("1.1", ["1.2.3"]) == "1.2.3.1"
-    assert _incremented_ver("1.1", ["1.2.3.4.5"]) == "1.2.3.4.6"
-    assert _incremented_ver("1.4.40", ["1.4.50"]) == "1.4.50.1"
-    assert _incremented_ver("1.4.0.40", ["1.4.0.50"]) == "1.4.0.50.1"
-    assert _incremented_ver("1.2.post3", ["1.2.3.4"]) == "1.2.3.5"
+    assert _stub_ver("1.1", ["1.2"]) == f"1.2.0.{STUB_V}"
+    assert _stub_ver("1.1", ["1.2.3"]) == f"1.2.3.{STUB_V}"
+    assert _stub_ver("1.2", ["1.3.0.4"]) == f"1.3.0.{STUB_V}"
+    assert _stub_ver("1.1", ["1.2.3.4.5"]) == f"1.2.3.4.{STUB_V}"
+    assert _stub_ver("1.4.40", ["1.4.50"]) == f"1.4.50.{STUB_V}"
+    assert _stub_ver("1.4.0.40", ["1.4.0.50"]) == f"1.4.0.50.{STUB_V}"
+    assert _stub_ver("1.2.post3", ["1.2.3.4"]) == f"1.2.3.{STUB_V}"
 
     # published less than version spec
-    assert _incremented_ver("1.2", ["1.1.0.4"]) == "1.2.0.0"
-    assert _incremented_ver("1", ["0.9"]) == "1.0.0.0"
-    assert _incremented_ver("1.1", ["0.9"]) == "1.1.0.0"
-    assert _incremented_ver("1.2.3", ["1.1.0.17"]) == "1.2.3.0"
-    assert _incremented_ver("1.2.3.4", ["1.1.0.17"]) == "1.2.3.4.0"
-    assert _incremented_ver("1.2.post3", ["1.1.0.17"]) == "1.2.3.0"
+    assert _stub_ver("1.2", ["1.1.0.4"]) == f"1.2.0.{STUB_V}"
+    assert _stub_ver("1", ["0.9"]) == f"1.0.0.{STUB_V}"
+    assert _stub_ver("1.1", ["0.9"]) == f"1.1.0.{STUB_V}"
+    assert _stub_ver("1.2.3", ["1.1.0.70"]) == f"1.2.3.{STUB_V}"
+    assert _stub_ver("1.2.3", [f"1.1.0.{STUB_V}"]) == f"1.2.3.{STUB_V}"
+    assert _stub_ver("1.2.3", [f"1.1.0.{FUTURE_V}"]) == f"1.2.3.{STUB_V}"
+    assert _stub_ver("1.2.3.4", ["1.1.0.17"]) == f"1.2.3.4.{STUB_V}"
+    assert _stub_ver("1.2.post3", ["1.1.0.17"]) == f"1.2.3.{STUB_V}"
+    assert _stub_ver("1.2.3", ["1.1.0.21000101"]) == f"1.2.3.{STUB_V}"
 
     # published equals version spec
-    assert _incremented_ver("1.1", ["1.1"]) == "1.1.0.1"
-    assert _incremented_ver("1.1", ["1.1.0.4"]) == "1.1.0.5"
-    assert _incremented_ver("1.1", ["1.1.3.4"]) == "1.1.3.5"
-    assert _incremented_ver("1.2.3.4", ["1.2.3.4.5"]) == "1.2.3.4.6"
-    assert _incremented_ver("1.2.3.4.5", ["1.2.3.4.5"]) == "1.2.3.4.5.1"
-
-    # test that we do the max version right
-    assert _incremented_ver("1.2", ["1.1.0.7", "1.2.0.7", "1.3.0.7"]) == "1.3.0.8"
```

Comment:
I think this test is still useful; is only coverage of code path with multiple published versions. RHS just needs to be tweaked